op-program/client/l1/oracle.go

```diff
@@ -0,0 +1,18 @@
+package l1
+
+import (
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type Oracle interface {
+	// HeaderByHash retrieves the block header with the given hash.
+	HeaderByHash(blockHash common.Hash) eth.BlockInfo
+
+	// TransactionsByHash retrieves the transactions from the block with the given hash.
+	TransactionsByHash(blockHash common.Hash) (eth.BlockInfo, types.Transactions)
+
+	// ReceiptsByHash retrieves the receipts from the block with the given hash.
+	ReceiptsByHash(blockHash common.Hash) (eth.BlockInfo, types.Receipts)
```

Comment:
Nit: maybe worth to clarify it's about block-hashes, not tx-hashes or other hashes.

```suggestion
	// HeaderByBlockHash retrieves the block header with the given hash.
	HeaderByBlockHash(blockHash common.Hash) eth.BlockInfo

	// TransactionsByBlockHash retrieves the transactions from the block with the given hash.
	TransactionsByBlockHash(blockHash common.Hash) (eth.BlockInfo, types.Transactions)

	// ReceiptsByBlockHash retrieves the receipts from the block with the given hash.
	ReceiptsByBlockHash(blockHash common.Hash) (eth.BlockInfo, types.Receipts)
```

* and change implementations / usages.